nat-lab/tests/utils/network_switcher/network_switcher_windows.py

```diff
@@ -41,17 +41,20 @@ def __init__(self, default: str, primary: str, secondary: str) -> None:
     async def create(connection: Connection) -> "ConfiguredInterfaces":
         interfaces = await Interface.get_network_interfaces(connection)
 
-        def find_interface(prefix: str) -> str:
+        def find_interface(prefix: str) -> Optional[str]:
             for interface in interfaces:
                 if interface.ipv4.startswith(prefix):
                     return interface.name
-            assert False, f"interface not found with prefix `{prefix}`, {interfaces}"
+            return None
 
-        return ConfiguredInterfaces(
-            find_interface(config.LIBVIRT_MANAGEMENT_NETWORK_PREFIX),
-            find_interface(config.PRIMARY_VM_NETWORK_PREFIX),
-            find_interface(config.SECONDARY_VM_NETWORK_PREFIX),
-        )
+        # Allow management interface to be shut down
+        management_itf = find_interface((config.LIBVIRT_MANAGEMENT_NETWORK_PREFIX))
```

Comment:
I think it would be a light improvement to assert on management interface being present if the target OS is Windows and MacOS

Reply:
Yeah, I was thinking about MacOS too, but on MacOS routing is different, and there is no competition with DHCP advertised routes versus windows. So right now I have changed only how windows is being configured.